src/routes/user.js

```diff
@@ -1,45 +1,112 @@
 import {Router} from 'express';
+import utils from '../utils';
 
 const router = Router();
 
+// User login.
+router.post('/login', async (req, res) => {
+	try {
+		const {username, password} = req.body;
+
+		const token = await req.context.models.User.findByLogin(username, password);
+
+		if (token.length > 0) {
+			return res.send(token);
+		}
+	} catch {
+		res.status(400).send('Invalid input');
+		return;
+	}
+});
+
+// Gets all users.
 router.get('/', async (req, res) => {
 	try {
-		const users = await req.context.models.User.findAll({
-			attributes: ['id','username','createdAt','updatedAt']
-		});
+		if (await utils.validateToken(req, res)) {
+			const users = await req.context.models.User.findAll({
+				attributes: ['username', 'email', 'phone', 'createdAt', 'updatedAt']
+			});
 
-		return res.send(users);
-	} catch (e) {
-		res.status(400).send(e);
+			return res.send(users);
+		}
+	} catch (error) {
+		res.status(400).send('Invalid input');
+		return;
 	}
 });
 
-router.get('/:userId', async (req, res) => {
+// Gets a specific user.
+router.get('/:username', async (req, res) => {
 	try {
-		const user = await req.context.models.User.findOne({
-			where: {
-				id: req.params.userId
-			},
-			attributes: ['id','username','token','createdAt','updatedAt']
-		});
-
-		return res.send(user);
-	} catch (e) {
-	 res.status(400).send(e);
- 	}
+		if (await utils.validateToken(req, res)) {
+			const user = await req.context.models.User.findOne({
+				where: {
+					username: req.params.username
+				},
+				attributes: ['username', 'email', 'phone', 'createdAt', 'updatedAt']
+			});
+
+			return res.send(user);
+		}
+	} catch {
+		res.status(400).send('Invalid payload');
+		return;
+	}
 });
 
-router.post('/login', async (req, res) => {
+// Creates a new user.
+router.post('/', async (req, res) => {
 	try {
-		const { username, password } = req.body;
-		
-		const response = await req.context.models.User.findByLogin(username,password);
-		console.log(response);
-
-		res.send(response);
-	} catch (e) {
-		res.status(400).send(e);
+		const {username, password} = req.body;
+
+		const user = await req.context.models.User.create({username, password});
+		return res.send(user.username + ' created');
+	} catch {
+		return res.status(400).send('Invalid input');
 	}
-})
+});
+
+// Updates any user.
+router.put('/', async (req, res) => {
```

Comment:
We should probably look into making an improvement on this one to tighten security so you can only update your own user account. Or add in account types so that admins can make changes in the system as well. Food for thought.

Reply:
I agree. This should be a future ticket though so we can add roles, etc.